[rocksdb] update to v6.13.3

kobykahane opened the pull request:
- Update port to latest upstream release.
- Rebase patches.
- Fix zstd feature.
- Add missing bzip2 feature.

NancyLi1013 commented on ports/rocksdb/0001-disable-gtest.patch:
```diff
@@ -1,13 +1,13 @@
++  #add_subdirectory(third-party/gtest-1.8.1/fused-src/gtest)
+   add_library(testharness STATIC
+   test_util/mock_time_env.cc
+   test_util/testharness.cc)
```
The format for this patch seems to be not right. The end of line should be `LF `instead of `CRLF`.

kobykahane replied:
Fixed.

NancyLi1013 reviewed (commented):
Have you tested these features? Can these work fine in this version?

kobykahane committed:
[rocksdb] update to v6.13.3

kobykahane commented:
> Have you tested these features? Can these work fine in this version?

I've tested consumption of [core,lz4] which is what my downstream project depends on. For the others, I've only tested compilation. In particular, the zstd feature was broken (before this PR) and I've now fixed it. Additionally, a feature for bzip2 was missing and I have now added it as well.

NancyLi1013 commented on ports/rocksdb/0005-backport-msvc-fixes-pr-7439.patch:
```diff
@@ -135,35 +239,18 @@ index 64cdb2f44..514943164 100644
++     defined(_M_IX86)) /* _mm_prefetch() is not defined outside of x86/x64 */
  #    include <mmintrin.h>   /* https://msdn.microsoft.com/fr-fr/library/84szxsww(v=vs.90).aspx */
  #    define XXH_PREFETCH(ptr)  _mm_prefetch((const char*)(ptr), _MM_HINT_T0)
  #  elif defined(__GNUC__) && ( (__GNUC__ >= 4) || ( (__GNUC__ == 3) && (__GNUC_MINOR__ >= 1) ) )
```
There are a lot of changes in this PR. Could you please submit this changes to upstream?

kobykahane replied:
I have already done so and that PR was already merged. This patch is a backport of https://github.com/facebook/rocksdb/pull/7439, which was already merged to upstream master, but is not present in their latest release tag (v6.13.3).

BillyONeal committed:
Merge remote-tracking branch 'origin/master' into HEAD

BillyONeal committed:
Add comment indicating which change is backported to make it easier to eliminate next version update.

BillyONeal committed:
Merge remote-tracking branch 'origin/master' into HEAD

BillyONeal commented:
Thanks for your contribution!
